### ISSUE TITLE
SendWebMessage from WindowCreated handler raises a System.AccessViolationException

### ISSUE THREAD

**Issue by albertospelta** [tryphotino/photino.NET#87]
Hi,
I've come across this System.AccessViolationException error after trying to send a web message from the WindowCreated handler.

After digging into the issue, I realized that this stems from the fact that the creation of the Photino.Native `<ICoreWebView2> _webviewWindow` instance is deferred until the message loop is started. Probably due to the `Photino::AttachWebView()` `CreateCoreWebView2EnvironmentWithOptions` callback.

**To reproduce:**

```csharp
var window = new PhotinoWindow
{
    StartUrl = "wwwroot/main.html",
    WindowCreatedHandler = (sender, args) => (sender as PhotinoWindow).SendWebMessage("test message"),
};
window.WaitForClose();
```
**Expected behavior**
I would expect all methods to be available in the WindowCreated event handler since the window has already been created

**Additional context**
- .NET 5, VisualStudio 2022
- Photino.NET package version 2.1.11
- Windows 11 version 21H2 build 22000

Great work on this framework!
Thanks,
Alberto

**Comment by ottodobretsberger** [comment]
Thank you for raising this issue. The way this would need to be addressed is to implement a callback event that notifies upon successful creation of the browser, after which the SendWebMessage will be available.
If time permits we will look into that as well, unless the community would like to pick up this task.

**Comment by SixFive7** [comment]
## Same Issue with `SetDevToolsEnabled()` - Workaround Found

We're experiencing the same issue with `SetDevToolsEnabled(true)` causing an `AccessViolationException` when called during window construction.

### Debugging Findings

1. **WebView2 isn't ready during construction** - The native `ICoreWebView2` instance doesn't exist until `WaitForClose()` starts the message pump and `CreateCoreWebView2EnvironmentWithOptions` completes.

2. **`RegisterWindowCreatedHandler` doesn't help** - We tried enabling DevTools in the `WindowCreatedHandler`, but WebView2 still isn't fully initialized at that point, causing the same crash.

3. **Main STA vs Secondary STA** - We run Photino on a dedicated STA thread (not the main `[STAThread]` thread). Some P/Invoke calls that work on the main STA thread cause access violations on secondary STA threads, even when properly configured with `Thread.SetApartmentState(ApartmentState.STA)`.

### Workaround: Wait for Page Ready

We wait for the page's JavaScript to signal readiness before calling `SetDevToolsEnabled`:

```javascript
// In our HTML page
window.chrome.webview.postMessage(JSON.stringify({ type: 'page:ready' }));
```

```csharp
// Only after receiving the ready message
_window.Invoke(() => _window.SetDevToolsEnabled(true)); // Now safe
```

By the time JavaScript executes, WebView2 is fully initialized and the call succeeds.

### Improved Workaround: Bridge Page Pattern

To avoid requiring every page to include the ready signal, we use a **bridge page** that:

1. Sends the ready signal once on load
2. Embeds the actual content via an iframe

```html
<!-- bridge.html - loaded once at startup -->
<!DOCTYPE html>
<html>
<head>
    <style>
        html, body, iframe { margin: 0; padding: 0; width: 100%; height: 100%; border: none; }
    </style>
</head>
<body>
    <iframe id="content" src="about:blank"></iframe>
    <script>
        // Signal ready immediately - WebView2 is now initialized
        window.chrome.webview.postMessage(JSON.stringify({ type: 'page:ready' }));

        // Listen for navigation commands from C#
        window.chrome.webview.addEventListener('message', (e) => {
            const msg = JSON.parse(e.data);
            if (msg.type === 'navigate') {
                document.getElementById('content').src = msg.url;
            }
        });
    </script>
</body>
</html>
```

This way:
- The bridge page is loaded once at startup
- It immediately signals ready, allowing deferred configuration like `SetDevToolsEnabled`
- All subsequent navigation happens inside the iframe
- No other pages need to know about this limitation

### Why This Is Still Not Ideal

- **Added complexity** - A bridge/wrapper page shouldn't be necessary
- **iframe limitations** - Some features don't work well inside iframes
- **Deferred configuration** - Settings that belong in the builder chain are delayed
- **Undocumented limitation** - It's not clear which methods work during construction vs. after `WaitForClose()`

### Suggested Fix

1. **Queue deferred calls** - Methods like `SetDevToolsEnabled` should queue their operations and execute after WebView2 initializes
2. **Expose a WebView Ready event** - Provide an official callback for when WebView2 is ready (distinct from `WindowCreatedHandler`)
3. **Document the limitation** - At minimum, clarify which methods require WebView2 to exist
